src/sugar3/graphics/popwindow.py

```diff
@@ -0,0 +1,245 @@
+# Copyright (C) 2016 Abhijit Patel
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# Some examples:
+#   Viewsource window
+#   Viewhelp window
+#
+# Can be further implemented in activities like
+#   Physics activity for Load Sample window.
+#   Read activity for the bookmark window.
+#   Bibliography for "new item" winodow.
+#   Pippy activity for Load example window.
+
+from gi.repository import Gtk
+from gi.repository import Gdk
+from gi.repository import GdkX11
+from gi.repository import GObject
+
+from sugar3.graphics import style
+from sugar3.graphics.toolbutton import ToolButton
+from sugar3.graphics.toolbarbox import ToolbarBox
+
+from jarabe.model import shell
+
+
+class PopWindow(Gtk.Window):
+    """
+    UI interface for activity Pop-up Windows.
+    PopWindows are the windows that open on the top of the current window.
+    These pop-up windows don't cover the whole screen.
+    They contain canvas content, alerts messages, a tray and a
+    toolbar.
+
+    FULLSCREEN and HALF_WIDTH for setting size of the window.
+
+    Kwargs:
+        set_titlebox (bool): False if titlebox is not needed
+        window_xid (xlib.Window): xid of the parent window
+    """
+    FULLSCREEN = (Gdk.Screen.width() - style.GRID_CELL_SIZE * 5,
+                  Gdk.Screen.height() - style.GRID_CELL_SIZE * 3)
+
+    HALF_WIDTH = ((Gdk.Screen.height() - style.GRID_CELL_SIZE * 5)/2,
+                  (Gdk.Screen.height() - style.GRID_CELL_SIZE * 3))
+
+    def __init__(self, set_titlebox=True, window_xid=None, **kwargs):
```

Comment:
Having just one prop would be very good developer experience:

```
popup = PopWindow(PopWindow.FULLSCREEN)
```

On Thu, Apr 21, 2016 at 7:55 AM, Abhijit Patel 
<notifications@github.com> wrote:
> In src/sugar3/graphics/popwindow.py:
> 
> > +    They contain canvas content, alerts messages, a tray and a
> > +    toolbar.
> > +
> > +    FULLSCREEN and HALF_WIDTH for setting size of the window.
> > +
> > +    Kwargs:
> > +        set_titlebox (bool): False if titlebox is not needed
> > +        window_xid (xlib.Window): xid of the parent window
> > +    """
> > +    FULLSCREEN = (Gdk.Screen.width() - style.GRID_CELL_SIZE * 5,
> > +                  Gdk.Screen.height() - style.GRID_CELL_SIZE * 3)
> > +
> > +    HALF_WIDTH = ((Gdk.Screen.height() - style.GRID_CELL_SIZE * 
> 5)/2,
> > +                  (Gdk.Screen.height() - style.GRID_CELL_SIZE * 3))
> > +
> > +    def __init__(self, set_titlebox=True, window_xid=None, 
> **kwargs):
> Adding props for size(having tuple of width, height) or separate for 
> width and height.
> I would prefer the separate properties. What are your suggestions 
> @samdroid-apps ?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>